docs/eventing/sugar/README.md

```diff
@@ -0,0 +1,358 @@
+
+```bash
+kubectl apply --filename {{< artifact repo="eventing" file="eventing-sugar-controller.yaml" >}}
+```
+
+## Automatic Broker Creation
+
+One way to create a Broker is to manually apply a resource to a cluster using
+the default settings:
+
+```shell
+kubectl create -f - <<EOF
+apiVersion: eventing.knative.dev/v1
+kind: Broker
+metadata:
+  name: default
+  namespace: default
+EOF
+```
+
+There might be cases where automated Broker creation is desirable, such as on
+namespace creation, or on Trigger creation. The Sugar controller enables those
+use-cases:
+
+- When a Namespace is labeled with `eventing.knative.dev/injection=enabled`, the
+  controller will create a default broker in that namespace.
+- When a Trigger is annotated with `eventing.knative.dev/injection=enabled`, the
+  controller will create a Broker named by that Trigger in the Trigger's
+  Namespace.
+
+When a Broker is deleted and the above labels or annotations are in-use, the
+Sugar Controller will automatically recreate a default Broker.
+
+### Namespace Examples
+
+Creating a "default" Broker when creating a Namespace:
+
+```shell
+kubectl apply -f - <<EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: example
+  labels:
+    eventing.knative.dev/injection: enabled
+EOF
+```
+
+To automatically create a Broker after a namespace exists, label the Namespace:
+
+```shell
+kubectl label namespace default eventing.knative.dev/injection=enabled
+```
+
+If the Broker named "default" already exists in the Namespace, the Sugar
+Controller will do nothing.
+
+### Trigger Examples
+
+Create a Broker named by a Trigger (`spec.broker`) in the Trigger's Namespace:
+
+```shell
+kubectl apply -f - << EOF
+apiVersion: eventing.knative.dev/v1
+kind: Trigger
+metadata:
+  name: hello-sugar
+  namespace: candyland
+  annotations:
+    eventing.knative.dev/injection: enabled
+spec:
+  broker: gumdrops
+  subscriber:
+    ref:
+      apiVersion: v1
+      kind: Service
+      name: event-display
+EOF
+```
+
+> _Note_: If the named Broker already exists, the Sugar controller will do
+> nothing, and the Trigger will not own the existing Broker.
+
+This will make a Broker called "gumdrops" in the Namespace "candyland", and
+attempt to send events to the "event-display" service.
+
+If the Broker of the given name already exists in the Namespace, the Sugar
+Controller will do nothing.
+
+## Automatic Trigger Creation
+
+The Sugar Controller can also watch for any resource that conforms to the duck
+type "Addressable", see [the definition](TODO: add a link for Addressable
+definition) or the [Discovery](TODO: link to the discovery documentation)
+ClusterDuckType
+[addressables.duck.knative.dev](https://github.com/knative-sandbox/discovery/blob/master/config/knative/addressables.duck.knative.dev.yaml).
+
+For any Addressable resource that has been labeled with
+`eventing.knative.dev/autotrigger=enabled`, the Sugar Controller will manage the
+Trigger(s) for it.
```

Comment:
Can this feature be combined with trigger-specified auto-broker creation? Such that the auto-created trigger also creates its broker?

Reply:
At the moment no. But I think that is an interesting idea so let me work it in.

Reply:
https://github.com/knative/docs/pull/3040/files#diff-a5afdc5012c7e77bf6fb45370fa61c4547612113c1aac0b1a8bf37500ae94f5fR369